### PR TITLE
Remove _PS_SMARTY_DIR_ pointing to no longer existing directory

### PR DESCRIPTION
**Pull request by mvorisek**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | see desc
| Type?             | bug fix
| Category?         | CO
| BC breaks?        | yes - `_PS_SMARTY_DIR_ ` constant was removed
| Deprecations?     | no
| Fixed ticket?     | n/a
| How to test?      | CI must pass
| Possible impacts? | no

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24634)

### CODE REVIEW

File: config/smarty.config.inc.php [hunk 1]
```diff
@@ -37,7 +36,6 @@
 $smarty->setCompileDir(_PS_CACHE_DIR_.'smarty/compile');
 $smarty->setCacheDir(_PS_CACHE_DIR_.'smarty/cache');
 $smarty->use_sub_dirs = true;
-$smarty->setConfigDir(_PS_SMARTY_DIR_.'configs');
```
Comment (mvorisek) [inline config/smarty.config.inc.php:40]: leftover from https://github.com/PrestaShop/TranslationToolsBundle/pull/39

/cc @Quetzacoalt91

**Review by eternoendless — DISMISSED** [review]
LGTM, but the breaking change must be noted

**Comment by eternoendless** [comment]
@mvorisek would you please note the removal of the constant as a breaking change?

**Comment by mvorisek** [comment]
@eternoendless please advise where or feel free to update the PR branch directly

**Comment by eternoendless** [comment]
In the BC breaks in the PR's description form. Please read [Describing Pull Requests](https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/pull-requests/)

**Comment by mvorisek** [comment]
done

**Review by Progi1984 — DISMISSED** [review]

**Comment by Progi1984** [comment]
@PrestaShop/qa-functional Hi, The PR removes a constant for Smarty. Do you need to check it ?

**Comment by eternoendless** [comment]
Context: https://www.smarty.net/docs/en/variable.config.dir.tpl

**Comment by Robin-Fischer-PS** [comment]
@Progi1984 Hi ! Can this fix have any impact (on installation or upgrade for example) ? If yes, we will do a check on the possible impacts, otherwise I don't think it's necessary to have QA on the PR.

**Comment by Progi1984** [comment]
@Robin-Fischer-PS I don't think so.

**Comment by Robin-Fischer-PS** [comment]
@Progi1984 Then no QA is needed. Thanks :D

**Comment by mvorisek** [comment]
UPDATE: I changed the Smarty conf. dir to:
```
$smarty->setConfigDir([]);
```

so no config is scanned in cwd at all.

ref:
- defautl: https://github.com/smarty-php/smarty/blob/4698dd9fb0406b0cee810e855b15818d517f9935/libs/Smarty.class.php#L617
- array support: https://github.com/smarty-php/smarty/blob/4698dd9fb0406b0cee810e855b15818d517f9935/libs/Smarty.class.php#L831

**Comment by mvorisek** [comment]
no QA, can we merge?

**Review by eternoendless — APPROVED** [review]
LGTM

**Comment by eternoendless** [comment]
@kpodemski what do you think? IMO this shouldn't have any impact unless people were putting smarty config files in `/vendor/prestashop/smarty` which sounds unlikely

**Comment by kpodemski** [comment]
@eternoendless only scenario I can remember related to Smarty, which can be somehow related, is support for registering custom modifiers, plugins, etc. but as you said, having them in `vendor/prestashop/smarty` is super unlikely

**Review by kpodemski — APPROVED** [review]

**Comment by eternoendless** [comment]
Thank you @mvorisek

**Pull request merged**